### PR TITLE
My-solution-task_dao_service

### PR DESCRIPTION
**Pull request by Swyatik19**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    public Driver create(Driver driver) {
+        String query = "INSERT INTO drivers (name,licenseNumber) VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, driver.getName());
+            statement.setString(2, driver.getLicenseNumber());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
```
Comment (karma-o) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:28]:
```suggestion
            ResultSet resultSet = statement.getGeneratedKeys();
```

File: src/main/java/mate/jdbc/Main.java [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
     public static void main(String[] args) {
-        // test your code here
+        Driver driver1 = new Driver(null, "bob", "123");
+        Driver driver2 = new Driver(null, "alice", "5678");
+        DriverService manager = (DriverService) injector.getInstance(DriverService.class);
+        manager.create(driver1);
+        manager.create(driver2);
+        System.out.println("List of all drivers:");
+        manager.getAll().forEach(System.out::println);
+
```
Comment (karma-o) [inline src/main/java/mate/jdbc/Main.java:19]: you should call each service method here

File: src/main/java/mate/jdbc/model/Driver.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@EqualsAndHashCode
+@ToString
+@RequiredArgsConstructor
```
Comment (karma-o) [inline src/main/java/mate/jdbc/model/Driver.java:13]: I think `Data` can replace all of these

**Review by karma-o — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Driver driver = null;
+            if (resultSet.next()) {
+                driver = getDriver(resultSet);
+            }
+            return Optional.ofNullable(driver);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a driver "
+                    + "from drivers table.", e);
```
Comment (Rommelua) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:52]:
```suggestion
            throw new DataProcessingException("Couldn't get a driver by id " + id, e);
```

File: src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE drivers SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() >= 1;
```
Comment (Rommelua) [inline src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java:96]:
```suggestion
            return statement.executeUpdate() > 0;
```

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**